draft-decraeneginsberg-lsr-isis-fast-flooding.xml

```diff
@@ -276,55 +274,57 @@
-			<t>Type: TBD1.</t>
-			<t>Length: variable, the size in octet of the Value field.</t>
-			<t>Value: a list of sub-TLVs.</t>
+			<t>This document defines a new Type-Length-Value tuple (TLV) called the "Flooding Parameters TLV" that may be included in IS to IS Hello (IIH) or Partial Sequence Number PDUs (PSNPs). It allows the LSP receiver to advertise flooding related parameters and capabilities and allows the LSP sender to better adapt to the receiver.</t>
+			<t>Type: TBD1</t>
+			<t>Length: variable, the size in octets of the Value field</t>
+			<t>Value: a list of sub-TLVs</t>
 			<t>Two sub-TLVs are defined in this document. The support of any sub-TLV is OPTIONAL.</t>
-			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The IS transmitter uses the latest received value for each parameter (sub-TLV) until a new value is advertised by the IS receiver. Note however that IIH and PSNP are not reliably exchanged, hence may never be received. For a parameter which has never been advertised, the IS transmitter use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
-			<section anchor="LSPReceptionWindow" title="InterfaceLSPReceiveWindow sub-TLV">
-				<t>The sub-TLV InterfaceLSPReceiveWindow advertises the maximum number of un-acknowledged LSPs that the node can receive/process with no separation interval between LSPs. This is meant to be the size of the LSP input queue of the IS-IS process in the control plane.</t>
-				<t>Type: 1.</t>
-				<t>Length: 4 octets.</t>
-				<t>Value: number of un-acknowledged LSPs which can be sent back to back.</t>
-				<t>Note that if an LSP has not been acknowledged and is sent again, it does not count twice. The reason is that this LSP is assumed to be lost and hence not in a buffer at the receiver.</t>
+			<t>For a given IS-IS adjacency, the Flooding Parameters TLV does not need to be advertised in each IIH or PSNP. The LSP transmitter uses the latest received value for each parameter until a new value is advertised by the LSP receiver. However, IIHs and PSNPs are not reliably exchanged, and may never be received, so parameters MAY be sent even if there is no change in value since the last transmission. For a parameter which has never been advertised, the LSP transmitter SHOULD use its local default value. That value SHOULD be configurable on a per node basis and MAY be configurable on a per interface basis.</t>
+			<section anchor="LSPReceptionWindow" title="LSP Burst Window sub-TLV">
+				<t>The LSP Burst Window sub-TLV advertises the maximum number of un-acknowledged LSPs that the node can receive with no separation interval between LSPs.</t>
+				<t>Type: 1</t>
+				<t>Length: 4 octets</t>
+				<t>Value: number of un-acknowledged LSPs that can be sent back to back</t>
```

Comment:
Remove the word "unacknowledged".
The sender would not send an LSP unless it has yet to receive an acknowledgment
(normal operation of Update Process).

Reply:
I disagree with Les' comment. From the LSP sender perspective, this parameter refers to the number of previously sent and still unacknowledged LSPs.

Reply:
It is very important that we all agree that there are no changes to the Update Process (as defined in ISO 10589) that are required or even suggested in order to support faster flooding. What this means is that which LSPs a Transmitter will send is still determined by the base Update Process - we are just trying to send them faster than before. A Transmitter would never send an LSP for which it has received an acknowledgement from the neighbor. To do so would represent a change in the Update Process. To use the word "unacknowledged" here suggests that there are two classes of LSPs that a transmitter may send: "Acknowledged LSPs" and "Unacknowledged LSPs". This is incorrect.

(I feel very strongly about this.)

Reply:
We agree that there is no change to the Update Process. I trust Tony will be able to propose some phrasing.
The term "unacknowledged" refers to the LSPs which have been already sent in the recent past, and not yet acknowledged (hence 'unacknowledged'). It does not refers to the LSP that will be sent.

Reply:
Transmitter will only send LSPs which are marked for transmission (SRM flag set in ISO 10589 terminology).
This does not change.
There is no need to tell the Transmitter that Burst only applies to Unacknowledged - unless you are proposing changes to the Retransmission interval. 

For me, the use of "Unacknowledged" is simply wrong.

Reply:
I'm very glad that we agree that there is no change in behavior.

The use of unacknowledged here is not wrong. The LSPs that are sent are unacknowledged. No correct transmitter would send an LSP that had been acknowledged. So the use of unacknowledged here is redundant and confusing. Removed.